sqlglot/dialects/snowflake.py

```diff
@@ -1277,3 +1285,8 @@ def select_sql(self, expression: exp.Select) -> str:
             if offset and not limit:
                 expression.limit(exp.Null(), copy=False)
             return super().select_sql(expression)
+
+        def put_sql(self, expression: exp.Put) -> str:
```

Comment:
Move this to the base generator, all expressions need to be generatable when using `sql()` without a dialect.

Reply:
Done